javascript/react-js/react-testing-part-one.md

```diff
@@ -0,0 +1,192 @@
+### Introduction
+
+We've learned testing our vanilla JavaScript applications in the previous section. Testing is indeed a powerful tool that allows to write maintainable and flexible code. If you've followed along our lessons, we've been using the [Jest](https://jestjs.io/) framework. For React, we'll keep using Jest and add more capabilities to our tests using the React Testing library.
+
+### Learning Outcomes
+
+By the end of this lesson, you should be able to do the following:
+
+* Discern what packages will be needed to test a React app.
+* Write queries to get elements from the UI and test them.
+* Be able to simulate user events with the `userEvent` API.
+* Explain what a snapshot test is and how to carry out a snapshot test with Jest.
```

Comment:
These aren't quite as high level as what I like to see. Rather than stating specifically what the student will be able to do our current direction for LOs is to outline the things that they'll learn about, i.e. "Test UI elements" vs "Write queries to get elements from the UI and test them."

In this [style guide update PR ](https://github.com/TheOdinProject/curriculum/pull/23051) we're moving towards having "Lesson Overviews" rather than even identifying them as "Learning Outcomes" to move towards this goal. Check the LO related content in that PR out and let me know if what I'm asking for makes sense.

Reply:
I've tried to make the LOs more of a general overview, though I'm still unsure with my changes. 

I've read through the style guide PR and I'm confused whether to change the "Learning Outcomes" heading to "Learning Overview". I've [commented on the style guide PR](https://github.com/TheOdinProject/curriculum/pull/23051#discussion_r791461163) for this matter.